openshift/console
# Bug 1826051: Support Cluster Trigger Binding

**Fixes**: 
https://issues.redhat.com/browse/ODC-3638

**Analysis / Root cause**: 
With the addition of 0.11 Pipeline Operator, the Trigger code for 4.5 no longer works (group change); also supports the Operator orientated ClusterTriggerBinding model.

**Solution Description**: 
Support ClusterTriggerBinding and fix up the code to use it.

**Screen shots / Gifs for design review**: 
![image](https://user-images.githubusercontent.com/8126518/79794375-24558d00-8320-11ea-9804-473a84580d13.png)

cc @openshift/team-devconsole-ux 

**Unit test coverage report**: 

Had to cut some bad tests out, can't be deep diving otherwise it makes brittle tests like this one.

**Test setup:**

* Install `canary` on the Community OpenShift Pipelines Operator (which should give you 0.11)
* Head over and create a Pipeline (oc create if you want)
* Add a Trigger to the Pipeline (note that the Operator installed some ClusterTriggerBindings, use one of those)
* Head over to the EventListener created (Pipeline Details => TriggerBinding Details => EventListener Details)
** You'll see it shows the CTB (ClusterTriggerBinding) resource now

**Browser conformance**:
